tests/ui/parser/missing-expression-in-for-loop.stderr

```diff
@@ -0,0 +1,13 @@
+error: expected expression, found block
```

Comment:
It is *marginally* misleading to say this, because blocks that have a tail expression *are* allowed, but I feel that it might be fine. If we were to change anything, I'd likely change this to
```suggestion
error: missing expression to iterate on in `for` loop
```
and maybe point at the space between `in` and the `{`. *But* that span could be potentially incorrect if that was indeed the iterating expression... 🤔 

I'm ok with this as is, I guess.